### PR TITLE
install: Ensure all formulae are linked

### PR DESCRIPTION
**Pull request by sjackman**
Link formulae that are unlinked and not keg only.

Fixes https://github.com/Homebrew/homebrew-bundle/issues/468

### CODE REVIEW

File: lib/bundle/brew_installer.rb [hunk 1]
```diff
@@ -75,6 +75,11 @@ def link_change_state!
       when false
         puts "Unlinking #{@name} formula." if ARGV.verbose?
```
Comment (MikeMcQuaid) [inline lib/bundle/brew_installer.rb]: Maybe this warrants a `linked?` check or similar (like below)

File: lib/bundle/brew_installer.rb [hunk 2]
```diff
@@ -75,6 +75,11 @@ def link_change_state!
...
         Bundle.system("brew", "unlink", @name)
+      when nil
+        if unlinked?
```
Comment (MikeMcQuaid) [inline lib/bundle/brew_installer.rb]: Does this verify keg-only state?

**Comment by vraravam** [comment]
@sjackman - could you please get the review comments fixed?

File: lib/bundle/brew_installer.rb (hunk 1 above)
Comment (sjackman) [reply lib/bundle/brew_installer.rb]: Done.

File: lib/bundle/brew_installer.rb (hunk 2 above)
Comment (sjackman) [reply lib/bundle/brew_installer.rb]: Yes.
https://github.com/Homebrew/homebrew-bundle/blob/d6e194364227dfdc11dff12d86d1dfee7b065f5f/lib/bundle/brew_dumper.rb#L119-L126

Comment (sjackman) [reply lib/bundle/brew_installer.rb]: I renamed this predicate `unlinked?` to `unlinked_and_not_keg_only?`.

File: lib/bundle/brew_installer.rb (hunk 1 above)
Comment (sjackman) [reply lib/bundle/brew_installer.rb]: It's not entirely straight forward with the data that's currently stored. `f[:link?]` stores either `true`, `false`, or `nil`.
- `true` indicates it's linked and keg-only.
- `false` indicates it's not linked and not keg-only.
- `nil` indicates its linkage is typical, either (linked and not keg-only) or (not linked and keg only).

Simply whether the formula is linked or not is not explicitly stored by `Bundle::BrewDumper`.

**Comment by sjackman** [comment]
> Looks good but can you write tests?

How do I run the tests in `spec/bundle/brew_installer_spec.rb`?

**Comment by sjackman** [comment]
`bundle exec rake` gave these errors with Ruby 2.5.1.
```
❯❯❯ bundle exec rake
/Users/sjackman/.homebrew/Cellar/ruby/2.5.1/lib/ruby/2.5.0/uri.rb:106:in `require': /Users/sjackman/.homebrew/Cellar/ruby/2.5.1/lib/ruby/2.5.0/uri/generic.rb:212: syntax error, unexpected '.' (SyntaxError)
      @scheme&.freeze
               ^
/Users/sjackman/.homebrew/Cellar/ruby/2.5.1/lib/ruby/2.5.0/uri/generic.rb:335: syntax error, unexpected '.'
      @scheme = v&.downcase
                   ^
/Users/sjackman/.homebrew/Cellar/ruby/2.5.1/lib/ruby/2.5.0/uri/generic.rb:1320: syntax error, unexpected '.'
      if path&.empty?
               ^
```
Is that expected?

File: lib/bundle/brew_installer.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply lib/bundle/brew_installer.rb]: @sjackman Could consider extending Homebrew/brew to provide the information here?

**Comment by MikeMcQuaid** [comment]
@sjackman I have no idea what's going on there. I'd suggest running with the normal Homebrew or system version of Ruby.

File: lib/bundle/brew_installer.rb (hunk 1 above)
Comment (sjackman) [reply lib/bundle/brew_installer.rb]: Never mind. I figured it out. Fixed in 0e0ba74ec025bb5b9e7b82fcfd15c7a312e2b16d

**Comment by sjackman** [comment]
I have the tests working with homebrewed Ruby 2.6.1p33. I'm not sure how much of the following was necessary, but it was sufficient.

```sh
brew upgrade ruby
$(brew --prefix ruby)/bin/gem install bundler -v1.17.3
$(brew --prefix ruby)/bin/gem install bundler
$(brew --prefix ruby)/bin/bundle install
$(brew --prefix ruby)/bin/bundle exec rake
```

**Comment by vraravam** [comment]
ping

**Comment by sjackman** [comment]
I'm defending my PhD thesis next week. Should have time after that to work on this PR again.

**Comment by MikeMcQuaid** [comment]
> I'm defending my PhD thesis next week. Should have time after that to work on this PR again.

@sjackman Good luck 🤞 ❤️

**Comment by MikeMcQuaid** [comment]
@sjackman Gentle bump on this! Shout if I can help at all.

**Comment by sjackman** [comment]
Success!

**Comment by MikeMcQuaid** [comment]
Thanks again @sjackman!

**Pull request merged**